libs/template/presets/src/features/fulfillment/components/fulfillment-root/fulfillment-root.component.ts

```diff
@@ -11,6 +16,35 @@ export class FulfillmentRootComponent extends LitElement {
 
   router = new LitRouter(this, this.extraRoutes ?? []);
 
+  constructor(
+    protected authService = resolve(OauthService),
+    protected routerService = resolve(RouterService)
+  ) {
+    super();
+  }
+
+  @subscribe()
+  protected guard$ = combineLatest([
+    this.authService.isAuthenticated(),
+    this.routerService.currentRoute(),
+  ]).pipe(
+    tap(([authenticated, currentRoute]) => {
+      if (
+        !authenticated &&
+        currentRoute !== '/login' &&
+        !currentRoute.startsWith('/oauth')
```

Comment:
Can you inject oauth config to get login url and oauth handler url instead of the hardcoded ones?